File: apps/frontend/features/summaries/lib/src/domain/entities/reader_action.dart
final class ReaderAction {
  const ReaderAction({
    required this.kind,
    required this.label,
    required this.reason,
    required this.citationIds,
    this.canonicalUrl,
  });

  final String kind;
  final String label;
  final String reason;
  final List<String> citationIds;
  final String? canonicalUrl;
}

File: libs/summary/domain/entities/reader-action.ts
export type ReaderActionKind =
  | "read_source"
  | "watch_repository"
  | "monitor_interest"
  | "compare_sources"
  | "ignore_low_confidence"
  | "add_interest_rule"
  | "request_deeper_scan"
  | "mark_relevant"
  | "mark_not_relevant";

export type ReaderAction = {
  readonly kind: ReaderActionKind;
  readonly label: string;
  readonly reason: string;
  readonly citationIds: readonly string[];
  readonly canonicalUrl?: string;
};
